subprojects/base-services/src/main/java/org/gradle/internal/exceptions/DefaultMultiCauseException.java

```diff
@@ -74,12 +74,14 @@ private void writeObject(java.io.ObjectOutputStream out) throws IOException {
     }
 
     private ThreadLocal<Boolean> threadLocal() {
-        return new ThreadLocal<Boolean>() {
-            @Override
-            protected Boolean initialValue() {
-                return false;
-            }
-        };
+        return new HideStacktrace();
+    }
+
+    private static class HideStacktrace extends ThreadLocal<Boolean> {
```

Comment:
We're limited to <Java 8, so a static inner class is the only way to replace this one.